File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/InMemorySecretRepositoryUnitTest.java
package at.tugraz.tc.cyfile.secret;

import junit.framework.Assert;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import at.tugraz.tc.cyfile.secret.impl.InMemorySecretRepository;

public class InMemorySecretRepositoryUnitTest {

    private InMemorySecretRepository memorySecretRepository;

    @Mock
    private Secret secret;


    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);

        memorySecretRepository = new InMemorySecretRepository(secret);

    }

    @Test
    public void getSecretShouldReturnSecretIfSecretSet() {
        // given secret is set
        memorySecretRepository = new InMemorySecretRepository(() -> "test");
        // when secret is retrieved
        Secret secret = memorySecretRepository.getSecret();

        // then secret value is correct
        Assert.assertEquals(secret.getSecretValue(), "test");
    }

    @Test
    public void getSecretShouldReturnNullIfSecretNotSet() {
        // given secret is not set
        memorySecretRepository = new InMemorySecretRepository(null);

        // when secret is retrieved
        Secret secret = memorySecretRepository.getSecret();

        // then secret value is null
        Assert.assertNull(secret);

    }

    @Test
    public void setSecretShouldSetSecretIfNotNull() {
        //given initialized state

        //when secret is being set
        boolean isSet = memorySecretRepository.saveSecret(() -> "secret");

        //then secret is saved
        Assert.assertTrue(isSet);

    }

    @Test
    public void setSecretShouldNotSetSecretIfNull() {
        //given initialized state

        //when null is being set
        boolean isSet = memorySecretRepository.saveSecret(null);

        //then secret is not saved
        Assert.assertFalse(isSet);
    }
}

File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/InMemorySecretRepositoryUnitTest.kt
package at.tugraz.tc.cyfile.secret

import at.tugraz.tc.cyfile.BaseUnitTest
import at.tugraz.tc.cyfile.secret.impl.InMemorySecretRepository
import at.tugraz.tc.cyfile.secret.impl.PinPatternSecret
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class InMemorySecretRepositoryUnitTest : BaseUnitTest() {

    private lateinit var memorySecretRepository: InMemorySecretRepository

    @Before
    fun setup() {
        memorySecretRepository = InMemorySecretRepository()

    }

    @Test
    fun getSecretShouldReturnSecretIfSecretSet() {
        // given secret is set
        memorySecretRepository = InMemorySecretRepository()
        memorySecretRepository.saveSecret(object : Secret {
            override val secretValue: String
                get() = "test"
        })

        // when secret is retrieved
        val secret = memorySecretRepository.retrieveSecret()!!

        // then secret value is correct
        Assert.assertEquals(secret.secretValue, "test")
    }

    @Test
    fun getSecretShouldReturnNullIfSecretNotSet() {
        // given secret is not set
        memorySecretRepository = InMemorySecretRepository()

        // when secret is retrieved
        val secret = memorySecretRepository.retrieveSecret()

        // then secret value is null
        Assert.assertNull(secret)

    }

    @Test
    fun setSecretShouldSetSecretIfNotNull() {
        //given initialized state

        //when secret is being set
        val isSet = memorySecretRepository.saveSecret(PinPatternSecret("secret"))

        //then secret is saved
        Assert.assertTrue(isSet)

    }
}

